src/Umbraco.Web.UI.Client/src/views/common/overlays/querybuilder/querybuilder.controller.js

```diff
@@ -154,7 +165,13 @@
                 });
 
         }, 200);
-        
+
+        function datePickerChange(event) {
+            templateQueryResource.postTemplateQuery(vm.query)
```

Comment:
The value was updated on ngModel but there were some timing issues which f***** it up. I have changed the code a bit so you now have to use the date which comes from the callback. If I understand you correctly, the code is now what you expected to see and not so much angular magic :)